pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -96,16 +87,45 @@ var heapsterMetricDefinitions = map[api.ResourceName]metricDefinition{
 		}},
 }
 
-func (h *HeapsterMetricsClient) ResourceConsumption(namespace string) ResourceConsumptionClient {
-	return &HeapsterResourceConsumptionClient{
-		namespace:           namespace,
-		client:              h.client,
+func NewHeapsterMetricsClient(client client.Interface) *HeapsterMetricsClient {
+	return &HeapsterMetricsClient{
+		client:              client,
 		resourceDefinitions: heapsterMetricDefinitions,
 	}
 }
 
-func (h *HeapsterResourceConsumptionClient) Get(resourceName api.ResourceName, selector map[string]string) (*extensions.ResourceConsumption, error) {
-	podList, err := h.client.Pods(h.namespace).
+func (h *HeapsterMetricsClient) GetCPUUtilization(namespace string, selector map[string]string) (*extensions.Utilization, error) {
+	usage, err := h.GetResourceConsumption(api.ResourceCPU, namespace, selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get CPU usage: %v", err)
+	}
+
+	// TODO: We are reading pod list twice - getting usage and here.
+	podList, err := h.client.Pods(namespace).
+		List(labels.SelectorFromSet(labels.Set(selector)), fields.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pod list: %v", err)
+	}
+	request := resource.MustParse("0")
+	for _, pod := range podList.Items {
+		for _, container := range pod.Spec.Containers {
+			containerRequest := container.Resources.Requests[api.ResourceCPU]
+			if containerRequest.Amount != nil {
+				request.Add(containerRequest)
+			}
+		}
+	}
+	if request.Cmp(resource.MustParse("0")) == 0 {
```

Comment:
I think this is somehow wrong. If *any* pod has missing request we shouldn't autoscale. (We should be more conservative).